packages/core/src/app/analytics/AnalyticsProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import { BodlService, CheckoutPaymentMethodExecutedOptions, CheckoutService, createBodlService, createStepTracker, StepTracker } from '@bigcommerce/checkout-sdk';
+import React, { ReactNode, useMemo } from 'react';
+import AnalyticsContext, { AnalyticsTrackerEvents } from './AnalyticsContext';
+import createAnalyticsService from './createAnalyticsService';
+
+interface AnalyticsProviderProps {
+    checkoutService: CheckoutService;
+    children: ReactNode;
+}
+
+const AnalyticsProvider = ({ checkoutService, children }: AnalyticsProviderProps) => {
+    const getStepTracker = useMemo(
+        () => createAnalyticsService<StepTracker>(createStepTracker, [checkoutService]),
+        [checkoutService]
+    );
```

Comment:
What downside if we initiate `stepTracker` here?

```suggestion
    const stepTracker = createAnalyticsService<StepTracker>(createStepTracker, [checkoutService]);
```

Reply:
The same for the bodl service

Reply:
Hi @solofeed 
If you ask why we can't to create analytics in such way: `const getStepTracker = createStepTracker(checkoutService);` ?

The reason is that in moment when React context initialisation, [checkoutService=>config](https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/analytics/create-step-tracker.ts#L33) hasn't initialised yet and we get [error and throw](https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/analytics/create-step-tracker.ts#L36) because of this.

Thats why I leave initialisation for analytics services on first call to this services, as [it was before.](https://github.com/bigcommerce/checkout-js/blob/master/packages/core/src/app/checkout/CheckoutApp.tsx#L78)